api/app/controllers/spree/api/v2/storefront/account/addresses_controller.rb

```diff
@@ -0,0 +1,69 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        module Account
+          class AddressesController < ::Spree::Api::V2::BaseController
+            before_action :require_spree_current_user
+
+            def index
+              render_serialized_payload { serialize_collection(collection) }
+            end
+
+            def create
+              result = create_service.call(user: spree_current_user, address_params: address_params)
+              render_result(result)
+            end
+
+            def update
+              result = update_service.call(address: resource, address_params: address_params)
+              render_result(result)
+            end
+
+            private
+
+            def collection
+              spree_current_user.addresses
```

Comment:
@kgorazd let's move this into a finder class and also add it as a dependency, it's a common pain point (eg. filter only addresses per kind etc) eg. https://github.com/spree/spree/blob/966386c200175473d04d3449718fd985a6910474/api/app/controllers/spree/api/v2/storefront/countries_controller.rb#L32